test/performance/odp_l2fwd.c

```diff
@@ -2035,13 +2045,23 @@ int main(int argc, char *argv[])
 		return -1;
 	}
 
-	pkt_len = POOL_PKT_LEN;
+	pkt_len = gbl_args->appl.packet_len;
 
 	if (pool_capa.pkt.max_len && pkt_len > pool_capa.pkt.max_len) {
 		pkt_len = pool_capa.pkt.max_len;
 		printf("\nWarning: packet length reduced to %u\n\n", pkt_len);
 	}
 
+	seg_len = pkt_len > MIN_SEG_LEN ? MIN_SEG_LEN : pkt_len;
+	if (seg_len < (pkt_len / pool_capa.pkt.max_segs_per_pkt))
+		seg_len = pkt_len / pool_capa.pkt.max_segs_per_pkt;
```

Comment:
Instead of this logic, would it be better to make segment length also a command line option?

Reply:
Adding segment length option will be good suggestion, will submit revised patch with new argument.
And I will move this logic to under no argument case, because if user requested bigger packet length(say 16K) and platform supports limited segments(assume 8), keeping default segment length (1536) will be problem here, so we need to tune segment length based on new packet length.